proposals/lambda-improvements.md

```diff
@@ -104,43 +104,60 @@ f = delegate int { return 1; };         // syntax error
 f = delegate int (int x) { return x; }; // syntax error
 ```
 
-## Natural delegate type
-A lambda expression has a natural type if the parameters types are explicit and either the return type is explicit or there is a common type from the natural types of all `return` expressions in the body.
+## Natural (function) type
+A lambda expression has a natural type if the parameters types are explicit and the return type is either explicit or can be inferred (see [inferred return type](https://github.com/dotnet/csharplang/blob/main/spec/expressions.md#inferred-return-type)).
 
-The natural type is a delegate type where the parameter types are the explicit lambda parameter types and the return type `R` is:
-- if the lambda return type is explicit, that type is used;
-- if the lambda has no return expressions, the return type is `void` or `System.Threading.Tasks.Task` if `async`;
-- if the common type from the natural type of all `return` expressions in the body is the type `R0`, the return type is `R0` or `System.Threading.Tasks.Task<R0>` if `async`.
+A method group has a natural type if all candidate methods in the method group have a common signature. (If the method group may include extension methods, the candidates include the containing type and all extension method scopes.)
 
-The natural type of an _individual method_ in a method group is a delegate type with the parameter types, ref kinds, and return type and ref kind, of that method. Parameter names and custom modifiers are ignored.
+The natural type of a lambda expression of method group is a _function_type_ that represents the method signature: the parameter types and ref kinds, and return type and ref kind.
+Lambda expressions or method groups with the same signature have the same _function_type_.
 
-A method group may contain multiple methods across the containing type and all extension methods.
-The natural type of the _method group_ is the common natural type for all methods in the method group.
-If there is no common type, the method group has no natural type.
+A _function_type_ exists at compile time only: _function_types_ do not appear in source or metadata, and are not available from the compiler API.
```

Comment:
> and are not available from the compiler API.

that seems restrictive.  what's the reason for that?  i can see potential scenarios where IDE woudl like to ask if two things have the same function-type.